examples/basic-usage.php

```diff
@@ -60,3 +60,17 @@ function () use ($event) {
 $event = new UserCreatedEvent('789', 'user@example.com');
 $future = $dispatcher->dispatch($event, new TimeoutCancellation(30));
 EventLoop::run();
+
+// Set up logging for your dispatcher - all errors will be logged to PSR logger
+$logger = new ColinODell\PsrTestLogger\TestLogger();
+$dispatcher = new AsyncEventDispatcher(
+    $listenerProvider,
+    $logger
```

Comment:
I think a error handler api would be a "cleaner" api. Consumers can decide what they want to do with the errors.

```
$dispatcher  = new AsyncEventDispatcher();
$dispatcher->setErrorHandler(function ($err) {
 // Implementors handle however they want to.
 Logger::exception($err); // For example
});
```

This library doesn't need to know anything about Psr\Log.

Reply:
Switch to using a closure.